File: tests/vb/vb_collections_advanced/array_declaration.vb
' vybe-test: vb/vb_collections_advanced/array_declaration
' origin: languages/vb/tests/vb/test_vb_collections_advanced.rs

' Vybe test harness — Visual Basic.
'
' Real VB source alongside harness/go/check.go and harness/js/check.js, the way
' test262's assert.js is JavaScript.
'
' A test's verdict is its EXIT CODE. __Check prints its diagnostic BEFORE
' throwing: an uncaught exception surfaces as `RuntimeError: [object]`, which
' says nothing at all.

Module VybeCheck
    Sub __Check(got As String, want As String)
        If got <> want Then
            Console.WriteLine("FAIL: want [" & want & "] got [" & got & "]")
            Throw New Exception("assertion failed")
        End If
    End Sub
End Module

Module M
    Sub Main()
        Dim arr(4) As Integer
        arr(0) = 10
        arr(1) = 20
        arr(2) = 30
        arr(3) = 40
        arr(4) = 50
        __Check(CStr(arr(2)), "30")
        __Check(CStr(UBound(arr)), "4")
    End Sub
End Module

File: tests/fortran/types/array_declaration.f90
! vybe-test: fortran/types/array_declaration
! origin: languages/fortran/tests/fortran/test_types.rs

program test
integer :: vybe_check_i = 0
integer :: vybe_check_w(1) = [ 30 ]
    integer, dimension(5) :: arr
    integer :: i
    do i = 1, 5
        arr(i) = i * 10
    end do
        vybe_check_i = vybe_check_i + 1
    if (vybe_check_i > 1) then
        print *, "FAIL: more than 1 line(s)"
        stop 1
    end if
    if ((arr(3)) /= vybe_check_w(vybe_check_i)) then
        print *, "FAIL at ", vybe_check_i, " got [", arr(3), "]"
        stop 1
    end if
if (vybe_check_i /= 1) then
    print *, "FAIL: ", vybe_check_i, " line(s), wanted 1"
    stop 1
end if
end program test
